utils/gym_utils.py

```diff
@@ -156,10 +156,8 @@ def _apply_safety(self, proposed_action: torch.Tensor, observation: torch.Tensor
             if self.safety_filter.p_ee_ref is None or not np.allclose(
                 self.safety_filter.p_ee_ref, goal_coords.reshape(-1, B)
             ):
-                nq = (observation.shape[1] - 3) // 2
-                q = observation[:, :nq]
                 self.safety_filter.set_reference_point(
```

Comment:
The same thing happens for the MPC as well right (inside predict_np of the CallableMPCExpert)? should we change it there too?

Reply:
Or perhaps it's not necessary since the n_seg of Observation Model (Estimator) and MPC are the same. But still might be more elegant and consistent that way.